src/controllers/foods.js

```diff
@@ -0,0 +1,122 @@
+import { response } from 'express';
+import { mysqlConnection } from '../database.js';
+import BaseSQLController from './base-sql.js';
+
+export class FoodsController extends BaseSQLController {
+	/**
+	 * @param {string} SingularEntityId The entity ID
+	 * @param {string} PluralEntityId The entity ID
+	 */
+	constructor(SingularEntityId, PluralEntityId) {
+		super(SingularEntityId ?? 'food', PluralEntityId ?? 'foods');
+		this.PluralEntityId = PluralEntityId ? PluralEntityId : 'foods';
+		this.SingularEntityId = SingularEntityId ? SingularEntityId : 'food';
+	}
+
+	//#region get methods
+
+	/**
+	 * Gets all foods
+	 * @param {Request} req The Express request
+	 * @param {Response} res The Express response
+	 */
+
+	getFoods(_req, res) {
+		const query = `SELECT * FROM foods`;
+		this.getAll(
+			query,
+			response => res.status(200).json(response),
+			error => res.status(500).json(error)
+		);
+	}
+
+	async getFoodByNameMethods(req, res) {
+		const query = 'SELECT * FROM foods WHERE food_name= ?';
+		const food_name = req;
+
+		return new Promise((resolve, reject) => {
+			mysqlConnection.query(query, [food_name], (error, rows, _fields) => {
+				if (!error) {
+					resolve(rows);
+				} else {
+					reject(
+						res.status(500).json({
+							message: 'Sorry we have an unexpected error trying fetch food by name',
+							error: user.sqlMessage
+						})
+					);
+				}
+			});
+		});
+	}
+
+	//#endregion
+
+	/**
```

Comment:
JSDoc to document your function should be done just before your function starts. Here is before a #region line and then you have your function

Reply:
Fixed in the last commit